Problem4/src/problem4/Ball.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package problem4;
+
+import java.util.Random;
+
+/**
+ *
+ * @author manxboy
+ */
+public class Ball {
```

Comment:
Your helper Ball/Vec2d classes should be implemented using OOP best practices.  Member variables should be private with appropriate getters and setters.  In the Ball class, your constructor could have also set the starting velocity (instead of setting it to a default value).